p2p/host/relaysvc/relay.go

```diff
@@ -65,14 +65,18 @@ func (m *RelayManager) background(ctx context.Context) {
 func (m *RelayManager) reachabilityChanged(r network.Reachability) error {
 	switch r {
 	case network.ReachabilityPublic:
+		m.mutex.Lock()
+		defer m.mutex.Unlock()
+		// ignore events with no reachability change
```

Comment:
```suggestion
		// This could happen if two consecutive EvtLocalReachabilityChanged report the same reachability.
		// This shouldn't happen, but it's safer to double-check.
```